data/api/client-server/relations.yaml

```
@@ -27,55 +27,303 @@ produces:
 securityDefinitions:
   $ref: definitions/security.yaml
 paths:
-  "/rooms/{roomId}/relations/{eventId}/{relType}/{eventType}":
+  "/rooms/{roomId}/relations/{eventId}":
+    get:
+      summary: Get the child events for a given parent event.
+      description: |-
+        Retrieve all of the child events for a given parent event.
+
+        Note that when paginating the `from` token must be *before* the `to` token, if
+        supplied. This means it's only possible to paginate "forwards" through events.
+        An example of an *invalid* call would be a `from` token from page 2 of the results,
+        and a `to` token from page 1: the server can't go backwards with this information.
+        The caller can use a `from` token from page 1 and a `to` token from page 2 to
+        paginate forwards over the same range, however.
```

I don't think that's *invalid*, per se, it's just that you're asking for the empty set.

It's not the fact that `from` must be before `to` that determines the fact that you can only go forwards; it's that directionality isn't implied by the relative orders of `from` and `to`. (Currently we *always* go forwards. MSC3715 changes that.)

Likewise below, x2

---

Sorry, I don't understand what needs to change here. If we swap "invalid" for "returns empty set", does that fix it?

---

That's a start!

We've also now discovered this endpoint goes backwards in time in its current, pre-MSC3715, implementation. Taking that into account:

```suggestion
        Note that when paginating the `from` token should be "after" the `to` token in
        terms of topological ordering, because it is only possible to paginate "backwards" 
        through events, starting at `from`.
        
        For example, passing a `from` token from page 2 of the results, and a `to` token
        from page 1, would return the empty set. The caller can use a `from` token from
        page 1 and a `to` token from page 2 to paginate over the same range, however.
```

I'm not sure the example helps, and I wonder if it would be clearer without.

---

that reads clear enough, I think. Certainly helps understand the problem at least :D

---

applied manually